crates/miden-objects/src/account/component/template/storage/toml.rs

```diff
@@ -576,6 +633,34 @@ fn parse_field_identifier<E: serde::de::Error>(
         })
 }
 
+/// Parses a `{ key, value }` TOML table into a `(Word, Word)` pair, rejecting templates.
+fn parse_map_entry_value(item: &toml::Value) -> Result<(Word, Word), InitStorageDataError> {
+    // Try to deserialize the user input as a map entry
+    let entry: MapEntry = MapEntry::deserialize(item.clone())
```

Comment:
It looks like we could pass the owned `toml::Value` to `flatten_parse_toml_value` and avoid the clone here.